### PR TITLE
OPT - Add functional tests to get repos for a project

### PR DESCRIPTION
**Pull request by CCardosoDev**

### CODE REVIEW

File: tests/functional/repos/test_repos.py [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+import mock
+import pytest
+
+
+class TestGetRepos:
+    @pytest.mark.django_db(transaction=True)
+    @pytest.mark.usefixtures('load_db_fixture')
+    @pytest.mark.parametrize(
+        'load_db_fixture', [{'fixture_filename': 'katka_project.json'}], indirect=True)
+    @mock.patch('requests.sessions.Session.request')
```
Comment (mjholtkamp) [inline tests/functional/repos/test_repos.py]: I see these decorators are repeated for the two tests, do you think they will be generic for all tests? in that case, how do you feel about moving some to the class instead?

File: tests/functional/repos/test_repos.py [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+        endpoint = reverse('repos', kwargs={'project_id': 'msap'})
+        response = client.get(
+            endpoint,
+            content_type='application/json',
+            data={'katka_project_id': '4cb432f5-0def-48b6-ad05-f1c082b1f1b8', 'limit': 10, 'start': 2}
+        )
+
+        assert len(response.data) == 2
+        assert response.data[0]['slug'] == response.data[0]['name']
+        assert response.data[1]['slug'] == response.data[1]['name']
```
Comment (mjholtkamp) [inline tests/functional/repos/test_repos.py]: How do you feel about testing an actual string. Now when the `slug` is null and the `name` is too and it's true for both, this test still succeeds, but I think that should be a failure.

If you test for `'invisible_women'` and `'katana'` then you can not have that problem.

File: bitbucket/views.py [hunk 3]
```diff
@@ -13,8 +13,16 @@ class BitbucketReposView(APIView):
 
     def get(self, request, project_id):
         data = request.data
-        data['project_id'] = project_id
-        data.update({'katka_project_id': request.query_params.get('katka_project_id')})
+        if request.query_params.get('limit') is not None:
```
Comment (rodgomes) [inline bitbucket/views.py:14]: minor: in the service there is already a check if the `limit` and `start` are None. You could skip the direct ifs here
```
data['limit'] = request.query_params.get('limit', None)
data['start'] = request.query_params.get('start', None)
```

File: tests/functional/repos/test_repos.py (hunk 1 above)
Comment (CCardosoDev) [reply tests/functional/repos/test_repos.py]: I have changed the class name to be more specific and moved the fixtures to class level

File: tests/functional/repos/test_repos.py (hunk 2 above)
Comment (CCardosoDev) [reply tests/functional/repos/test_repos.py]: Agree, added that test case also.

File: bitbucket/views.py (hunk 3 above)
Comment (CCardosoDev) [reply bitbucket/views.py:14]: I am doing this since None is not a valid value for integer serializer field.

Comment (rodgomes) [reply bitbucket/views.py:14]: ah, ok then!

**Review by mjholtkamp — APPROVED** [review]

**Pull request merged**